src/app/components/Search/SearchForm.jsx

```diff
@@ -1,56 +1,132 @@
 import React from 'react';
-import Store from '../../stores/Store';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import SearchButton from '../Button/SearchButton';
+import { search } from '../../actions/SearchActions';
 
-class SearchForm extends React.Component {
+export class SearchForm extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = Store.getState();
+    this.state = props;
 
-    this.handleChange = this.handleChange.bind(this);
+    this.onFilterChange = this.onFilterChange.bind(this);
+    this.onQueryChange = this.onQueryChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
+    this.submitSearchRequest = this.submitSearchRequest.bind(this);
   }
 
-  handleChange(event) {
-    console.log(event);
+  onFilterChange(event) {
+    const filterSelected = event.target.value;
+    this.setState({ filter: filterSelected });
+  }
+
+  onQueryChange(event) {
+    this.setState({ query: event.target.value });
   }
 
   handleSubmit(event) {
-    console.log(event);
+    if (event && event.charCode === 13) {
+      this.submitSearchRequest(event);
+    }
+  }
+
+  submitSearchRequest(event) {
+    event.preventDefault();
+    if (!this.state.filter) {
+      this.setState({ filter: 'q' });
+    }
+
+    this.props.search(this.state.query, this.state.filter);
   }
 
   render() {
+    const { searchResults, query, filter, allowedFilters, sort } = this.state;
```

Comment:
You're not using `searchResults` in this component so it might be good to remove here (and any other variable that's not being used).